client-src/elements/utils.ts

```diff
@@ -130,13 +122,11 @@ export function unambiguousStageName(
 
   // Count the stages of the same type that appear before this one. This is O(n^2) when it's used to
   // number every stage, but the total number of stages is generally <20.
-  const index = feature.stages
-    .filter(s => s.stage_type === stage.stage_type)
-    .findIndex(s => s.id === stage.id);
+  const stagesOfThisType = Object.groupBy(feature.stages, (s: StageDict) => s.stage_type)[stage.stage_type] || [];
```

Comment:
I think the chained method calls and use of .filter() was actually more clear.  This seems like an example of AI finding a place to do an upgrade that we asked for, but it doesn't really improve over the existing code.  Maybe we could provide an additional criteria as part of the prompt, e.g., "only use groupBy() when more than one of the resulting groups are read."